### PR TITLE
adding service account to libvirt

### PR DESCRIPTION
**Pull request by sdminonne**

### CODE REVIEW

**Comment by sdminonne** [comment]
@zmerlynn have you time for this?

**Comment by sdminonne** [comment]
@lavalamp @nikhiljindal may you have a look at this (feel free to reassign)?

**Comment by lavalamp** [comment]
Or maybe @liggitt?

File: cluster/libvirt-coreos/util.sh [hunk 1]
```diff
@@ -51,6 +52,19 @@ function detect-nodes {
   KUBE_NODE_IP_ADDRESSES=("${NODE_IPS[@]}")
 }
 
+function set_service_accounts {
+    SERVICE_ACCOUNT_KEY=${SERVICE_ACCOUNT_KEY:-"/tmp/kube-serviceaccount.key"}
```
Comment (liggitt) [inline cluster/libvirt-coreos/util.sh]: will /tmp be writeable under normal circumstances?

File: cluster/libvirt-coreos/util.sh [hunk 2]
```diff
@@ -51,6 +52,19 @@ function detect-nodes {
...
+    # Generate ServiceAccount key if needed
+    if [[ ! -f "${SERVICE_ACCOUNT_KEY}" ]]; then
+      mkdir -p "$(dirname ${SERVICE_ACCOUNT_KEY})"
+      openssl genrsa -out "${SERVICE_ACCOUNT_KEY}" 2048 2>/dev/null
```
Comment (liggitt) [inline cluster/libvirt-coreos/util.sh]: will openssl be available under normal circumstances?

File: cluster/libvirt-coreos/util.sh [hunk 3]
```diff
@@ -51,6 +52,19 @@ function detect-nodes {
...
+    fi
+
+    mkdir -p  "$POOL_PATH/kubernetes/certs"
```
Comment (liggitt) [inline cluster/libvirt-coreos/util.sh]: I'm not familiar enough with this install to know if this path is reasonable

File: cluster/libvirt-coreos/util.sh [hunk 4]
```diff
@@ -51,6 +52,19 @@ function detect-nodes {
...
+    cp ${SERVICE_ACCOUNT_KEY} "$POOL_PATH/kubernetes/certs"
```
Comment (liggitt) [inline cluster/libvirt-coreos/util.sh]: `"${SERVICE_ACCOUNT_KEY}"`

File: cluster/libvirt-coreos/util.sh [hunk 5]
```diff
@@ -22,7 +22,8 @@ source "$ROOT/${KUBE_CONFIG_FILE:-"config-default.sh"}"
 source "$KUBE_ROOT/cluster/common.sh"
 
 export LIBVIRT_DEFAULT_URI=qemu:///system
-
+export SERVICE_ACCOUNT_LOOKUP=${SERVICE_ACCOUNT_LOOKUP:-false}
+export ADMISSION_CONTROL=${ADMISSION_CONTROL:-NamespaceLifecycle,NamespaceAutoProvision,LimitRanger,ServiceAccount,ResourceQuota}
```
Comment (liggitt) [inline cluster/libvirt-coreos/util.sh]: I think NamespaceAutoProvision is deprecated (@derekwaynecarr, confirm?), and should probably be removed from local-up-cluster.sh as well.
@pmorie I see SecurityContextDeny in some config-default.sh scripts and not others... is it expected to be included by default or not?

File: cluster/libvirt-coreos/util.sh (hunk 1 above)
Comment (sdminonne) [inline cluster/libvirt-coreos/util.sh]: This env is local dev environment. /tmp in general is writable (as in local cluster)

File: cluster/libvirt-coreos/util.sh (hunk 4 above)
Comment (sdminonne) [inline cluster/libvirt-coreos/util.sh]: ok

File: cluster/libvirt-coreos/util.sh (hunk 3 above)
Comment (sdminonne) [inline cluster/libvirt-coreos/util.sh]: `$POOL_PATH` is the shared directory mounted by different virtual hosts data will be in `kubernetes/cluster/libvirt-coreos/libvirt_storage_pool` and mounted remotely in `/opt/`

File: cluster/libvirt-coreos/util.sh (hunk 5 above)
Comment (sdminonne) [inline cluster/libvirt-coreos/util.sh]: Thanks, I was not aware of all this.

Comment (derekwaynecarr) [inline cluster/libvirt-coreos/util.sh]: NamespaceAutoProvision is deprecated, it should be removed.

**Comment by derekwaynecarr** [comment]
@lhuard1A can you take a pass on these changes as well?

**Comment by sdminonne** [comment]
@derekwaynecarr: removed NamespaceAutoProvision and  @liggitt's feedback on double quoting. Squashed.

Waiting @lhuard1A feedback

File: cluster/libvirt-coreos/util.sh (hunk 2 above)
Comment (lhuard1A) [inline cluster/libvirt-coreos/util.sh]: `libvirt-coreos` targets local development environments.
I think it is reasonable to request `openssl` as a pre-requisite. It should be easy for the developer to install it on its workstation.
But we could document this new dependency in: https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/libvirt-coreos.md#prerequisites

File: cluster/libvirt-coreos/util.sh (hunk 3 above)
Comment (lhuard1A) [inline cluster/libvirt-coreos/util.sh]: In `libvirt-coreos` setup, the `$POOL_PATH/kubernetes` directory of the host (machine where `kube-up.sh` is launched) is mounted on each VM in `/opt/kubernetes`.

It is the most efficient way to push things to the VMs.

I’m fine with that.

File: cluster/libvirt-coreos/util.sh (hunk 2 above)
Comment (sdminonne) [inline cluster/libvirt-coreos/util.sh]: ok

**Comment by lhuard1A** [comment]
@derekwaynecarr : I had a look at @sdminonne’s change.
Everything looks reasonable to me.

**Comment by sdminonne** [comment]
This failure is a flake. @liggitt  @derekwaynecarr  as soon you're OK may you re-run the tests? Thanks

**Comment by liggitt** [comment]
@k8s-bot test this
@k8s-bot unit test this

**Comment by sdminonne** [comment]
Reopened to have re-run tests

**Comment by sdminonne** [comment]
@derekwaynecarr @liggitt can you help me with this? I'd like to have this merged

**Comment by derekwaynecarr** [comment]
@sdminonne - LGTM, tagged, and marked e2e-not-required get through the queue.

**Comment by sdminonne** [comment]
@derekwaynecarr  thanks

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**